sklearn/preprocessing/data.py

```diff
@@ -1956,3 +1956,66 @@ def transform(self, X):
         """
         return _transform_selected(X, self._transform,
                                    self.categorical_features, copy=True)
+
+
+class CountFeaturizer(object):
+
+    def __init__(self, data=None, inclusion=None):
+        if data != None:
+            self.fit(data, inclusion=inclusion)
+
+    def get_data(self):
+        return self.data 
+
+    def valid_data_type(self, type_check):
```

Comment:
generally these sort of checks are directly implemented in the methods that call them (so `fit` in this case)